modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/PublishAPIApplication.testcase

```diff
@@ -0,0 +1,225 @@
+		}
+
+		task ("Then I can get the publish status in response") {
+			var actualValue = JSONUtil.getWithJSONPath(${response}, "$.items..applicationStatus.name");
+
+			TestUtils.assertEquals(
+				actual = ${actualValue},
+				expected = "published");
+		}
+	}
+
+	@priority = 4
+	test CanGetUpdatedTitleInResponseAfterPublish {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+
+		task ("And Given I edit the title and publish the api application again") {
+			APIBuilderUI.goToEditAPIApplicationPage(key_title = "App-test");
+
+			Type(
+				locator1 = "APIBuilder#ENTER_TITLE",
+				value1 = "App-test-1");
+
+			Click(locator1 = "APIBuilder#PUBLISH_BUTTON");
+		}
+
+		task ("When with getAPIApplicationsPage to retrieve the api application") {
+			var response = ApplicationAPI.getAPIApplications();
+		}
+
+		task ("Then I can get the update title in response") {
+			var actualValue = JSONUtil.getWithJSONPath(${response}, "$.items..title");
+
+			TestUtils.assertEquals(
+				actual = ${actualValue},
+				expected = "App-test-1");
+		}
+	}
+
+	@priority = 4
+	test CanPromptCancelChangesDialog {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+
+		task ("And Given I edit the title of the API application") {
+			APIBuilderUI.goToEditAPIApplicationPage(key_title = "App-test");
+
+			Type(
+				locator1 = "APIBuilder#ENTER_TITLE",
+				value1 = "App-test-1");
+		}
+
+		task ("When I click Cancel button") {
+			Click(locator1 = "Button#CANCEL");
+		}
+
+		task ("Then I can see the Cancel Changes dialog with message and Continue Without Saving button") {
+			AssertTextEquals.assertPartialText(
+				locator1 = "Modal#BODY",
+				value1 = "Are you sure you want to continue without saving?");
+
+			AssertElementPresent(locator1 = "APIBuilder#CONTINUE_WITHOUT_SAVING_BUTTON");
+		}
+	}
+
+	@priority = 4
+	test CanPublishAPIApplication {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+
+		task ("When I click Publish button in edit application wizard") {
+			APIBuilderUI.publishApplication(key_title = "App-test");
+		}
+
+		task ("Then the status of the application changed to published") {
+			AssertTextEquals(
+				deployStatus = "Published",
+				locator1 = "AppBuilder#DEPLOY_APP_STATUS",
+				value1 = "Published");
+		}
+	}
+
+	@priority = 4
+	test CanPublishAPIApplicationWithChangePublicationStatus {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+
+		task ("When I click Change Publication Status") {
+			APIBuilderUI.changePublicationStatus(key_title = "App-test");
+		}
+
+		task ("Then the status of the application changed to published") {
+			AssertElementPresent(
+				key_status = "Published",
+				key_title = "App-test",
+				locator1 = "APIBuilder#APPLICATION_STATUS");
+		}
+	}
+
+	@priority = 4
+	test CanUpdateTitleAfterRePublish {
```

Comment:
This is not same with our original test case "Given I publish an API application". And if you'd like to update, please make sure the test steps are not missing. I mean we miss one step to publish the application before edit the title

Reply:
sorry, in my understanding to create an Application, I will modify it, thanks